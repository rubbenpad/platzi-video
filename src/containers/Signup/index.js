import React from 'react';
import { connect } from 'react-redux';

import { signupRequest } from '../../actions';
import googleIcon from '../../assets/google-icon.svg';
import twitterIcon from '../../assets/twitter-icon.svg';
import backIcon from '../../assets/back-icon.png';
import {
  Wrapper,
  Container,
  Form,
  Input,
  Button,
  Register,
  Option,
  Line,
  SLink,
} from './styles';

const Signup = (props) => {
  const [form, setForm] = React.useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
  });

  const handleChange = (event) => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    props.signupRequest(form);
    props.history.push('/');
  };

  return (
    <Wrapper>
      <SLink to="/">
        <img src={backIcon} alt="Back button" />
      </SLink>
      <Container>
        <Option>
          <img src={googleIcon} alt="Google Icon" />
          <span>Continue with Google</span>
        </Option>
        <Option>
          <img src={twitterIcon} alt="Google Icon" />
          <span>Continue with Twitter</span>
        </Option>
        <Line>
          <span>or</span>
        </Line>
        <Form onSubmit={handleSubmit}>
          <Input
            name="firstName"
            type="text"
            placeholder="First name"
            value={form.firstName}
            onChange={handleChange}
          />
          <Input
            name="lastName"
            type="text"
            placeholder="Last name"
            value={form.lastName}
            onChange={handleChange}
          />
          <Input
            name="email"
            type="email"
            placeholder="Email address"
            value={form.email}
            onChange={handleChange}
          />
          <Input
            name="password"
            type="password"
            placeholder="Create a password"
            value={form.password}
            onChange={handleChange}
          />
          <Button type="submit">Sing up</Button>
        </Form>
      </Container>
      <Register>
        <span>Already have an account?</span>
        <a href="/login">Log in</a>
      </Register>
    </Wrapper>
  );
};

const mapDispatchToProps = {
  signupRequest,
};

export default connect(null, mapDispatchToProps)(Signup);
