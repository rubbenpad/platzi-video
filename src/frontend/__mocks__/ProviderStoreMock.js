import React from 'react'
import { createStore, compose, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { Router } from 'react-router'
import { createBrowserHistory } from 'history'
import { Provider } from 'react-redux'
import reducer from '../reducers'
import initialState from '../initialState'

const store = createStore(
  reducer,
  initialState,
  compose(applyMiddleware(thunk))
)
const history = createBrowserHistory()

const ProviderStoreMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>{children}</Router>
  </Provider>
)

export default ProviderStoreMock
