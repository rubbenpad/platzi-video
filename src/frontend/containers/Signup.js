import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { registerUser } from '../actions'
import Form from '../components/Form'
import Input from '../components/Input'
import { Div, Button, RegisterBox } from './styles'

const Signup = props => {
  const [form, setForm] = React.useState({
    name: '',
    email: '',
    password: '',
  })

  const handleChange = event => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    })
  }

  const handleSubmit = event => {
    event.preventDefault()
    props.registerUser(form, '/signin')
  }

  return (
    <>
      <Div>
        <Form onSubmit={handleSubmit}>
          <Input
            name="name"
            type="text"
            placeholder="Name"
            value={form.name}
            onChange={handleChange}
            required
          />
          <Input
            name="email"
            type="email"
            placeholder="Email address"
            value={form.email}
            onChange={handleChange}
            required
          />
          <Input
            name="password"
            type="password"
            placeholder="Create a password"
            value={form.password}
            onChange={handleChange}
            required
          />
          <Button type="submit">Sing up</Button>
        </Form>
        <RegisterBox>
          <span>Already have an account?</span>
          <a href="/login">Log in</a>
        </RegisterBox>
      </Div>
    </>
  )
}

Signup.propTypes = {
  registerUser: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  registerUser,
}

export default connect(null, mapDispatchToProps)(Signup)
