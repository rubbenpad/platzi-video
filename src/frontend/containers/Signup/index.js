import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { registerUser } from '../../actions'
import {
  Wrapper,
  Container,
  Form,
  Input,
  Button,
  Register,
  StyledLink,
  BackButton,
} from './styles'

const Signup = props => {
  const [form, setForm] = React.useState({
    name: '',
    email: '',
    password: '',
  })

  const handleChange = event => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    })
  }

  const handleSubmit = event => {
    event.preventDefault()
    props.registerUser(form, '/signin')
  }

  return (
    <Wrapper>
      <StyledLink to="/">
        <BackButton />
      </StyledLink>
      <Container>
        <Form onSubmit={handleSubmit}>
          <Input
            name="name"
            type="text"
            placeholder="Name"
            value={form.name}
            onChange={handleChange}
            required
          />
          <Input
            name="email"
            type="email"
            placeholder="Email address"
            value={form.email}
            onChange={handleChange}
            required
          />
          <Input
            name="password"
            type="password"
            placeholder="Create a password"
            value={form.password}
            onChange={handleChange}
            required
          />
          <Button type="submit">Sing up</Button>
        </Form>
      </Container>
      <Register>
        <span>Already have an account?</span>
        <a href="/login">Log in</a>
      </Register>
    </Wrapper>
  )
}

Signup.propTypes = {
  registerUser: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  registerUser,
}

export default connect(null, mapDispatchToProps)(Signup)
