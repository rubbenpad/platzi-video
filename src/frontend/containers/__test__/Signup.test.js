import React from 'react'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { create } from 'react-test-renderer'
import ProviderStoreMock from '../../__mocks__/ProviderStoreMock'
import Signup from '../Signup'

configure({ adapter: new Adapter() })

describe('Sumbit form on signup test', () => {
  test('Match Snapshot', () => {
    const signup = create(
      <ProviderStoreMock>
        <Signup />
      </ProviderStoreMock>
    )
    expect(signup.toJSON()).toMatchSnapshot()
  })

  it('Calls and executes preventDefault fucntion on submit form', () => {
    const preventDefault = jest.fn()
    const wrapper = mount(
      <ProviderStoreMock>
        <Signup />
      </ProviderStoreMock>
    )
    wrapper.find('form').simulate('submit', { preventDefault })
    expect(preventDefault).toHaveBeenCalledTimes(1)
    wrapper.unmount()
  })
})
