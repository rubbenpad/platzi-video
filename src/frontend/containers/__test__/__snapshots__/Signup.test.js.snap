// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sumbit form on signup test Match Snapshot 1`] = `
<div
  className="sc-bxivhb ccFUwH"
>
  <form
    className="sc-bdVaJa eFjEm"
    onSubmit={[Function]}
  >
    <input
      className="sc-bwzfXH gUqduo"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      required={true}
      type="text"
      value=""
    />
    <input
      className="sc-bwzfXH gUqduo"
      name="email"
      onChange={[Function]}
      placeholder="Email address"
      required={true}
      type="email"
      value=""
    />
    <input
      className="sc-bwzfXH gUqduo"
      name="password"
      onChange={[Function]}
      placeholder="Create a password"
      required={true}
      type="password"
      value=""
    />
    <button
      className="sc-ifAKCX inDuyN"
      type="submit"
    >
      Sing up
    </button>
  </form>
  <div
    className="sc-EHOje cHxLlX"
  >
    <span>
      Already have an account?
    </span>
    <a
      href="/login"
    >
      Log in
    </a>
  </div>
</div>
`;
